app/celery/provider_tasks.py

```diff
@@ -43,6 +44,7 @@ def deliver_sms(self, notification_id):
         send_to_providers.send_sms_to_provider(notification)
     except Exception as e:
         try:
+            dao_toggle_sms_provider()
```

Comment:
I agree. But at the moment the `SMSClientResponseException` is thrown when we do not understand the response from the provider. But when we get timeout exceptions we get throw a different type of exception. Or there could be an unknown reason for the exception. So it will not work to catch `SMSClientResponseException`, without some refactoring.